localstack/utils/bootstrap.py

```diff
@@ -236,68 +223,56 @@ def get_docker_image_details(image_name=None):
     return result
 
 
-def get_docker_container_names():
-    cmd = "%s ps --format '{{.Names}}'" % config.DOCKER_CMD
-    try:
-        output = to_str(run(cmd))
-        container_names = re.split(r"\s+", output.strip().replace("\n", " "))
-        return container_names
-    except Exception as e:
-        LOG.info('Unable to list Docker containers via "%s": %s' % (cmd, e))
-        return []
-
-
 def get_main_container_ip():
     container_name = get_main_container_name()
-    cmd = "%s inspect -f '{{range .NetworkSettings.Networks}}{{.IPAddress}}{{end}}' %s" % (
-        config.DOCKER_CMD,
-        container_name,
-    )
-    return run(cmd, print_error=False).strip()
+    return DOCKER_CLIENT.get_container_ip(container_name)
 
 
 def get_main_container_id():
     container_name = get_main_container_name()
     try:
-        cmd = "%s inspect -f '{{ .Id }}' %s" % (config.DOCKER_CMD, container_name)
-        return run(cmd, print_error=False).strip()
-    except Exception:
+        return DOCKER_CLIENT.get_container_id(container_name)
+    except ContainerException:
         return None
 
 
 def get_main_container_name():
     global MAIN_CONTAINER_NAME_CACHED
     if MAIN_CONTAINER_NAME_CACHED is None:
         hostname = os.environ.get("HOSTNAME")
-        cmd = "%s inspect -f '{{ .Name }}' %s" % (config.DOCKER_CMD, hostname)
-        try:
-            MAIN_CONTAINER_NAME_CACHED = run(cmd, print_error=False).strip().lstrip("/")
-        except Exception:
+        if hostname:
+            try:
+                MAIN_CONTAINER_NAME_CACHED = DOCKER_CLIENT.get_container_name(hostname)
+            except ContainerException:
+                MAIN_CONTAINER_NAME_CACHED = config.MAIN_CONTAINER_NAME
+        else:
             MAIN_CONTAINER_NAME_CACHED = config.MAIN_CONTAINER_NAME
     return MAIN_CONTAINER_NAME_CACHED
 
 
 def get_server_version():
-    docker_cmd = config.DOCKER_CMD
     try:
         # try to extract from existing running container
         container_name = get_main_container_name()
-        version = run(
-            "%s exec -it %s bin/localstack --version" % (docker_cmd, container_name),
-            print_error=False,
+        version, _ = DOCKER_CLIENT.exec_in_container(
+            container_name, interactive=True, command=["bin/localstack", "--version"]
         )
-        version = version.strip().split("\n")[-1]
+        version = version.decode(config.DEFAULT_ENCODING).strip().splitlines()[-1]
```

Comment:
nit: we could use the `to_str(..)` util function here:
```
         version = to_str(version).strip().splitlines()[-1]
```
(also on line 273 below)